src/components/ConfigChecker.tsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+
+import { Button } from 'hds-react';
+import { Link, useLocation } from 'react-router-dom';
+import { getConfigByType, getConfigTypeFromRoute } from '../config';
+import { getClientConfig, setClientConfig } from '../client';
+import ConfigSelection from '../pages/ConfigSelector';
+import PageContent from './PageContent';
+import { getClient } from '../client/oidc-react';
+import { tunnistamoPath, keycloakPath } from '../paths';
+
+const ConfigChecker = (
+  props: React.PropsWithChildren<unknown>
+): React.ReactElement => {
+  const { children } = props;
+  const currentConfig = getClientConfig();
+  const location = useLocation();
+  const configType = getConfigTypeFromRoute(location.pathname);
+  if (!currentConfig && !configType) {
+    return <ConfigSelection />;
+  }
+  const configSetFromConfigType =
+    !currentConfig && configType
+      ? setClientConfig(getConfigByType(configType))
+      : undefined;
+  const activeConfig = configSetFromConfigType || currentConfig;
+  const isCurrentConfigForKeycloak = activeConfig.isKeycloakConfig;
```

Comment:
Many names and also logic in this file (and in some other files too) suggest the there is the _normal_ case and then there's the _keycloak_ case and the latter is somehow special or exceptional. But to me it seems it shouldn't be like that. There are just two different client configurations. At most one of them is in use at any given time and it's possible to swap between them. I would like to see this code written in such a way that it suggests that there are many (and many could be two, three or more) configurations, one of which is active.

Reply:
Configs contain more settings, so this kind of "isKeycloak" checks are not used anymore.